internal/generate/crd/crd.go

```diff
@@ -0,0 +1,351 @@
+				return nil, err
+			}
+			if typeMeta.Kind != "CustomResourceDefinition" {
+				continue
+			}
+			crd := &apiextv1beta1.CustomResourceDefinition{}
+			if err = yaml.Unmarshal(sb, crd); err != nil {
+				return nil, err
+			}
+			// controller-tools inserts an annotation and assumes that the binary
+			// that creates the CRD is controller-gen. In this case, we don't use
+			// controller-gen. Instead, we vendor and use the same library that
+			// controller-gen does.
+			//
+			// The value that gets populated in the annotation is based on the
+			// build info of the compiled binary, not on the version of the
+			// vendored controller-tools library.
+			//
+			// See: https://github.com/kubernetes-sigs/controller-tools/issues/348
+			//
+			// TODO(joelanford): Sort out what to do with this. Until then, let's
+			// just remove it.
+			delete(crd.Annotations, "controller-gen.kubebuilder.io/version")
+			if sb, err = k8sutil.GetObjectBytes(crd, yaml.Marshal); err != nil {
+				return nil, err
+			}
+			modifiedCRD = yamlutil.CombineManifests(modifiedCRD, sb)
+		}
+		if len(modifiedCRD) != 0 {
+			fileNameNoExt := strings.TrimSuffix(info.Name(), filepath.Ext(info.Name()))
+			fileMap[fileNameNoExt+"_crd.yaml"] = modifiedCRD
+		}
+	}
+	return fileMap, nil
+}
+
+// generateNonGo generates a CRD for non-Go projects using a resource.
+// There are currently no commands to update CRD manifests for non-Go
+// operators, so if a CRD manifest already exists for a resource, this
+// generator only checks and sorts CRD versions (for now).
+func (g crdGenerator) generateNonGo() (map[string][]byte, error) {
+	crd := &apiextv1beta1.CustomResourceDefinition{}
+	fileMap := map[string][]byte{}
+	fileName := getFileNameForResource(g.resource)
+	path := filepath.Join(g.inputDir, fileName)
+	if _, err := os.Stat(path); err == nil {
+		b, err := ioutil.ReadFile(path)
+		if err != nil {
+			return nil, err
+		}
+		if err = yaml.Unmarshal(b, crd); err != nil {
+			return nil, err
+		}
+		// If version is new, append it to spec.versions.
+		hasVersion := false
+		for _, version := range crd.Spec.Versions {
+			if version.Name == g.resource.Version {
+				hasVersion = true
+				break
+			}
+		}
+		if !hasVersion {
+			// Let either the user or below logic determine whether this new
+			// version is stored or not.
+			crd.Spec.Versions = append(crd.Spec.Versions, apiextv1beta1.CustomResourceDefinitionVersion{
+				Name:    g.resource.Version,
+				Storage: false,
+				Served:  true,
+			})
+		}
+	} else if os.IsNotExist(err) {
+		crd = newCRDForResource(g.resource)
+	} else {
+		return nil, err
+	}
+
+	sort.Sort(k8sutil.CRDVersions(crd.Spec.Versions))
+	setCRDStorageVersion(crd)
+	if err := checkCRDVersions(crd); err != nil {
+		return nil, err
+	}
+	b, err := k8sutil.GetObjectBytes(crd, yaml.Marshal)
+	if err != nil {
+		return nil, err
+	}
+	fileMap[fileName] = b
+	return fileMap, nil
+}
+
+// newCRDForResource constructs a barebones CRD using data in resource.
```

Comment:
Everything in this file below this line was copied verbatim from the [CRD scaffold](https://github.com/operator-framework/operator-sdk/blob/460022f/internal/scaffold/crd.go#L189).

Reply:
This function needs to be updated based on #2275